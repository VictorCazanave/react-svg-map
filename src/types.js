import PropTypes from 'prop-types';

export const MapType = PropTypes.shape({
  viewBox: PropTypes.string.isRequired,
  locations: PropTypes.arrayOf(
    PropTypes.shape({
      path: PropTypes.string.isRequired,
      name: PropTypes.string,
      id: PropTypes.string,
    })
  ).isRequired,
  label: PropTypes.string,
});

export const LocationActions = {
  onLocationMouseOver: PropTypes.func,
  onLocationMouseOut: PropTypes.func,
  onLocationMouseMove: PropTypes.func,
  onLocationFocus: PropTypes.func,
  onLocationBlur: PropTypes.func,
};

export const LocationAttributes = {
  locationClassName: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
  locationAriaLabel: PropTypes.func,
};

export const MapAttributes = {
  childrenBefore: PropTypes.node,
  childrenAfter: PropTypes.node,
  className: PropTypes.string,
};

export const LocationType = PropTypes.shape({
  id: PropTypes.string.isRequired,
  name: PropTypes.string,
});
