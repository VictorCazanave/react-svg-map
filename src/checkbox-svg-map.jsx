import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import SVGMap from './svg-map';

class CheckboxSVGMap extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			// TODO: Store only ids instead of nodes?
			selectedLocations: []
		};

		this.isLocationSelected = this.isLocationSelected.bind(this);
		this.handleLocationClick = this.handleLocationClick.bind(this);
		this.handleLocationKeyDown = this.handleLocationKeyDown.bind(this);
	}

	componentDidMount() {
		// Set initial selected locations
		if (this.props.selectedLocationIds) {
			// TODO: Find a cleaner way
			// Cannot use ref on SvgMap (with React 16.0.0) because it is a functional component
			// https://5a046bf5a6188f4b8fa4938a--reactjs.netlify.app/docs/refs-and-the-dom.html#refs-and-functional-components
			const svgNode = ReactDOM.findDOMNode(this);
			const selectedLocations = this.props.selectedLocationIds.map(locationId => svgNode.getElementById(locationId))
				.filter(location => !!location); // Remove null locations when invalid id

			this.setState({ selectedLocations });
		}
	}

	/**
	 * Indicate whether a location is selected
	 * 
	 * @param {Object} location - Location object
	 * @returns {boolean} True if the location is selected
	 */
	isLocationSelected(location) {
		return this.state.selectedLocations.some(selectedLocation => selectedLocation.id === location.id);
	}

	/**
	 * Select/deselect a location
	 * 
	 * @param {Event} event - Triggered event
	 */
	toggleLocation(event) {
		const location = event.target;

		this.setState(prevState => {
			// Copy old state
			let selectedLocations = [...prevState.selectedLocations];

			if (location.attributes['aria-checked'].value === 'true') {
				// Delete location
				selectedLocations.splice(selectedLocations.indexOf(location), 1);
			} else {
				// Add location
				selectedLocations.push(location);
			}

			// Call onChange event handler
			if (this.props.onChange) {
				this.props.onChange(selectedLocations);
			}

			// Return new state
			return { selectedLocations };
		});
	}

	/**
	 * Handle click on a location
	 * 
	 * @param {Event} event - Triggered click event
	 */
	handleLocationClick(event) {
		event.preventDefault();
		this.toggleLocation(event);
	}

	/**
	 * Handle spacebar down on a location
	 * 
	 * @param {Event} event - Triggered keydown event
	 */
	handleLocationKeyDown(event) {
		// Spacebar
		if (event.keyCode === 32) {
			event.preventDefault();
			this.toggleLocation(event);
		}
	}

	render() {
		return (
			<SVGMap
				map={this.props.map}
				role="group"
				locationRole="checkbox"
				locationTabIndex="0"
				className={this.props.className}
				locationClassName={this.props.locationClassName}
				locationAriaLabel={this.props.locationAriaLabel}
				isLocationSelected={this.isLocationSelected}
				onLocationClick={this.handleLocationClick}
				onLocationKeyDown={this.handleLocationKeyDown}
				onLocationMouseOver={this.props.onLocationMouseOver}
				onLocationMouseOut={this.props.onLocationMouseOut}
				onLocationMouseMove={this.props.onLocationMouseMove}
				onLocationFocus={this.props.onLocationFocus}
				onLocationBlur={this.props.onLocationBlur}
				childrenBefore={this.props.childrenBefore}
				childrenAfter={this.props.childrenAfter}
			/>
		);
	}
}

CheckboxSVGMap.propTypes = {
	selectedLocationIds: PropTypes.arrayOf(PropTypes.string),
	onChange: PropTypes.func,

	// SVGMap props
	map: PropTypes.shape({
		viewBox: PropTypes.string.isRequired,
		locations: PropTypes.arrayOf(
			PropTypes.shape({
				path: PropTypes.string.isRequired,
				name: PropTypes.string,
				id: PropTypes.string
			})
		).isRequired,
		label: PropTypes.string
	}).isRequired,
	className: PropTypes.string,
	locationClassName: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
	locationAriaLabel: PropTypes.func,
	onLocationMouseOver: PropTypes.func,
	onLocationMouseOut: PropTypes.func,
	onLocationMouseMove: PropTypes.func,
	onLocationFocus: PropTypes.func,
	onLocationBlur: PropTypes.func,
	childrenBefore: PropTypes.node,
	childrenAfter: PropTypes.node,
};

export default CheckboxSVGMap;
