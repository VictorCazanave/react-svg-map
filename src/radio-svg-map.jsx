import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import SVGMap from './svg-map';

class RadioSVGMap extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			selectedLocation: null
		};

		this.getLocationTabIndex = this.getLocationTabIndex.bind(this);
		this.isLocationSelected = this.isLocationSelected.bind(this);
		this.handleLocationClick = this.handleLocationClick.bind(this);
		this.handleLocationKeyDown = this.handleLocationKeyDown.bind(this);
	}

	componentDidMount() {
		// List of location nodes
		// TODO: Find a cleaner way
		// Cannot use ref on SvgMap (with React 16.0.0) because it is a functional component
		// https://5a046bf5a6188f4b8fa4938a--reactjs.netlify.app/docs/refs-and-the-dom.html#refs-and-functional-components
		this.locations = [...ReactDOM.findDOMNode(this).getElementsByTagName('path')];

		// Set initial selected location
		if (this.props.selectedLocationId) {
			const selectedLocation = this.locations.find(location => location.id === this.props.selectedLocationId);

			this.setState({ selectedLocation });
		}
	}

	/**
	 * Get location tabindex value
	 * 
	 * @param {Object} location - Location object
	 * @param {number} index - Index of location
	 * @returns {string} Value of tabindex HTML attribute
	 */
	getLocationTabIndex(location, index) {
		let tabIndex = null;

		if (this.state.selectedLocation) {
			// Only selected location is focusable
			tabIndex = this.isLocationSelected(location) ? '0' : '-1';
		} else {
			// Only first location is focusable
			tabIndex = index === 0 ? '0' : '-1';
		}

		return tabIndex;
	}

	/**
	 * Indicate whether a location is selected
	 *
	 * @param {Object} location - Location object
	 * @returns {boolean} True if the location is selected
	 */
	isLocationSelected(location) {
		return this.state.selectedLocation && this.state.selectedLocation.id === location.id;
	}

	/**
	 * Select a location
	 * 
	 * @param {Node} location - Location DOM node
	 */
	selectLocation(location) {
		// Focus new selected location
		location.focus();

		// Change selected location
		this.setState({ selectedLocation: location });

		// Call onChange event handler
		if (this.props.onChange) {
			this.props.onChange(location);
		}
	}

	/**
	 * Handle click on a location
	 *
	 * @param {Event} event - Triggered click event
	 */
	handleLocationClick(event) {
		const clickedLocation = event.target;

		// Select clicked location if not already selected
		if (clickedLocation !== this.state.selectedLocation) {
			this.selectLocation(clickedLocation);
		}
	}

	/**
	 * Handle spacebar and arrows down on a location
	 *
	 * @param {Event} event - Triggered keydown event
	 */
	handleLocationKeyDown(event) {
		const focusedLocation = event.target;

		// Spacebar
		if (event.keyCode === 32) {
			event.preventDefault();

			// Select focused location if not already selected
			if (focusedLocation !== this.state.selectedLocation) {
				this.selectLocation(focusedLocation);
			}

			// Arrow down or right
		} else if (event.keyCode === 39 || event.keyCode === 40) {
			event.preventDefault();

			// Select next or first location
			this.selectLocation(focusedLocation.nextSibling || this.locations[0]);

			// Arrow up or left
		} else if (event.keyCode === 37 || event.keyCode === 38) {
			event.preventDefault();

			// Select previous or last location
			this.selectLocation(focusedLocation.previousSibling || this.locations[this.locations.length - 1]);
		}
	}

	render() {
		return (
			<SVGMap
				map={this.props.map}
				role="radiogroup"
				locationTabIndex={this.getLocationTabIndex}
				locationRole="radio"
				className={this.props.className}
				locationClassName={this.props.locationClassName}
				locationAriaLabel={this.props.locationAriaLabel}
				isLocationSelected={this.isLocationSelected}
				onLocationClick={this.handleLocationClick}
				onLocationKeyDown={this.handleLocationKeyDown}
				onLocationMouseOver={this.props.onLocationMouseOver}
				onLocationMouseOut={this.props.onLocationMouseOut}
				onLocationMouseMove={this.props.onLocationMouseMove}
				onLocationFocus={this.props.onLocationFocus}
				onLocationBlur={this.props.onLocationBlur}
				onChange={this.props.onChange}
				childrenBefore={this.props.childrenBefore}
				childrenAfter={this.props.childrenAfter}
			/>
		);
	}
}

RadioSVGMap.propTypes = {
	selectedLocationId: PropTypes.string,
	onChange: PropTypes.func,

	// SVGMap props
	map: PropTypes.shape({
		viewBox: PropTypes.string.isRequired,
		locations: PropTypes.arrayOf(
			PropTypes.shape({
				path: PropTypes.string.isRequired,
				name: PropTypes.string,
				id: PropTypes.string
			})
		).isRequired,
		label: PropTypes.string
	}).isRequired,
	className: PropTypes.string,
	locationClassName: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
	locationAriaLabel: PropTypes.func,
	onLocationMouseOver: PropTypes.func,
	onLocationMouseOut: PropTypes.func,
	onLocationMouseMove: PropTypes.func,
	onLocationFocus: PropTypes.func,
	onLocationBlur: PropTypes.func,
	childrenBefore: PropTypes.node,
	childrenAfter: PropTypes.node,
};

export default RadioSVGMap;
