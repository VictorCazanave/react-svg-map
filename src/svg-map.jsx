import React from 'react';
import PropTypes from 'prop-types';

function SVGMap(props) {
	return (
		<svg
			xmlns="http://www.w3.org/2000/svg"
			viewBox={props.map.viewBox}
			className={props.className}
			role={props.role}
			aria-label={props.map.label}
		>
			{props.childrenBefore}
			{props.map.locations.map((location, index) => {
				return (
					<path
						id={location.id}
						name={location.name}
						d={location.path}
						className={typeof props.locationClassName === 'function' ? props.locationClassName(location, index) : props.locationClassName}
						tabIndex={typeof props.locationTabIndex === 'function' ? props.locationTabIndex(location, index) : props.locationTabIndex}
						role={props.locationRole}
						aria-label={typeof props.locationAriaLabel === 'undefined' ? location.name : props.locationAriaLabel(location, index)}
						aria-checked={props.isLocationSelected && props.isLocationSelected(location, index)}
						onMouseOver={props.onLocationMouseOver}
						onMouseOut={props.onLocationMouseOut}
						onMouseMove={props.onLocationMouseMove}
						onClick={props.onLocationClick}
						onKeyDown={props.onLocationKeyDown}
						onFocus={props.onLocationFocus}
						onBlur={props.onLocationBlur}
						key={location.id}
					/>
				);
			})}
			{props.childrenAfter}
		</svg>
	);
}

SVGMap.propTypes = {
	// Map properties
	map: PropTypes.shape({
		viewBox: PropTypes.string.isRequired,
		locations: PropTypes.arrayOf(
			PropTypes.shape({
				path: PropTypes.string.isRequired,
				id: PropTypes.string.isRequired,
				name: PropTypes.string
			})
		).isRequired,
		label: PropTypes.string
	}).isRequired,
	className: PropTypes.string,
	role: PropTypes.string,

	// Locations properties
	locationClassName: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
	locationTabIndex: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
	locationRole: PropTypes.string,
	onLocationMouseOver: PropTypes.func,
	onLocationMouseOut: PropTypes.func,
	onLocationMouseMove: PropTypes.func,
	onLocationClick: PropTypes.func,
	onLocationKeyDown: PropTypes.func,
	onLocationFocus: PropTypes.func,
	onLocationBlur: PropTypes.func,
	isLocationSelected: PropTypes.func,
	locationAriaLabel: PropTypes.func,

	// Slots
	childrenBefore: PropTypes.node,
	childrenAfter: PropTypes.node,
};

SVGMap.defaultProps = {
	className: 'svg-map',
	role: 'none', // No role for map
	locationClassName: 'svg-map__location',
};

export default SVGMap;
