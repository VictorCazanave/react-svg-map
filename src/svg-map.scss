.svg-map {
	width: 100%;
	height: auto;
	stroke: #666;
	stroke-width: 1;
	stroke-linecap: round;
	stroke-linejoin: round;

	&__location {
		fill: #a1d99b;
		cursor: pointer;

		&:focus,
		&:hover {
			fill: #b8e2b3;
			outline: 0;
		}

		&[aria-checked=true] {
			fill: #f4bc44;
		}
	}
}
