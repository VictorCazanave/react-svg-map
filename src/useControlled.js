import { useState, useEffect, useRef, useCallback } from 'react';

export const useControlled = ({
  controlledValue,
  initialValue,
  name = 'SVG Map',
  onChange,
}) => {
  const isControlled = controlledValue !== undefined;
  const { current: origIsControlled } = useRef(isControlled);
  const [internalValue, setInternalValue] = useState(initialValue);
  const { current: origInitialValue } = useRef(initialValue);
  const value = origIsControlled ? controlledValue : internalValue;

  useEffect(() => {
    if (origIsControlled !== isControlled) {
      console.warn(
        `"${name}" is changed from ${
          origIsControlled
            ? 'uncontrolled to controlled'
            : 'controlled to uncontrolled'
        }.`
      );
    }
  }, [origIsControlled, isControlled, name]);

  useEffect(() => {
    if (!origIsControlled && origInitialValue !== initialValue) {
      console.warn(
        `"${name}" is changing its defaultValue after being initialized. Make "${name}" a controlled component instead.`
      );
    }
  }, [origInitialValue, initialValue, name, origIsControlled]);

  useEffect(() => {
    if (typeof onChange === 'function' && controlledValue === undefined) {
      console.warn(
        '* Please provide a "value" prop to your controlled Map component'
      );
    } else if (typeof onChange === undefined && controlledValue !== undefined) {
      console.warn(
        '* Please provide an "onChange" function to your controlled Map component'
      );
    }
  }, [controlledValue, onChange]);

  const setValueIfUncontrolled = useCallback(
    newValue => {
      if (!origIsControlled) {
        setInternalValue(newValue);
      }
    },
    [origIsControlled]
  );

  return [value, setValueIfUncontrolled, { isControlled }];
};
