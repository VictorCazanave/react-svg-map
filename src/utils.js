export function getNodeAttributes(node) {
	return Object.values(node.attributes).reduce(
		(acc, curr) => ({ ...acc, [curr.name]: curr.value }),
		{}
	);
}
