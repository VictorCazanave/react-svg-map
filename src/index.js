import SVGMap from './svg-map';
import Australia from './maps/australia';
import France from './maps/france';
import Taiwan from './maps/taiwan';
import USA from './maps/usa';
import Utah from './maps/utah';
import './svg-map.scss';

export {
	SVGMap,
	Australia,
	France,
	Taiwan,
	USA,
	Utah
};
