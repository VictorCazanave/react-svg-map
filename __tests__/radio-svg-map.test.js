import React from 'react';
import ReactDOM from 'react-dom';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';

import FakeMap from './fake-map';
import { RadioSVGMap } from '../src';

// TODO: Try to make it more readable
// TODO: Create utility functions to avoid code duplication
describe('RadioSVGMap component', () => {
	const locationSelector = 'id1';
	const previousLocationSelector = 'id0';
	const nextLocationSelector = 'id2';

	let wrapper;
	let location;
	let previousLocation;
	let nextLocation;

	describe('Navigation', () => {
		beforeEach(() => {
			wrapper = render(<RadioSVGMap map={FakeMap} />);
			location =  screen.getByTestId(locationSelector);
			previousLocation = screen.getByTestId(previousLocationSelector);
			nextLocation = screen.getByTestId(nextLocationSelector);
		});

		afterEach(() => {
			wrapper.unmount();
		});

		describe('Mouse', () => {
			it('selects location when clicking on not yet selected location', () => {
				expect(location.getAttribute('aria-checked')).toBe(null);

				userEvent.click(location);
				location = screen.getByTestId(locationSelector);

				expect(location).toHaveAttribute('aria-checked');
			});

			it('does not deselect location when clicking on already selected location', () => {
				userEvent.click(location);
				location = screen.getByTestId(locationSelector);

				expect(location).toHaveAttribute('aria-checked', 'true');

				userEvent.click(location);
				location = screen.getByTestId(locationSelector);

				expect(location).toHaveAttribute('aria-checked', 'true');
			});

			it('selects new location and deselects former selected when clicking on new location', () => {
				userEvent.click(location);
				location = screen.getByTestId(locationSelector);

				expect(location).toHaveAttribute('aria-checked', 'true');
				expect(previousLocation).toHaveAttribute('aria-checked', 'false');
				
				userEvent.click(previousLocation);
				location = screen.getByTestId(locationSelector);
				previousLocation = screen.getByTestId(previousLocationSelector);

				expect(location).toHaveAttribute('aria-checked', 'false');
				expect(previousLocation).toHaveAttribute('aria-checked', 'true');
			});

			it('makes location focusable when selected', () => {
				expect(location).toHaveProperty('tabIndex', -1);
				
				userEvent.click(location);
				location = screen.getByTestId(locationSelector);
				
				expect(location).toHaveProperty('tabIndex', 0);
			});
		});

		describe('Keyboard', () => {
			it('selects focused not yet selected location when hitting spacebar', () => {
				expect(location.getAttribute('aria-checked')).toBe(null);

				location.focus();
				userEvent.keyboard('{space}');
				location = screen.getByTestId(locationSelector);

				expect(location).toHaveAttribute('aria-checked', 'true');
			});

			it('does not deselect focused already selected location when hitting spacebar', () => {
				location.focus();
				userEvent.keyboard('{space}');
				location = screen.getByTestId(locationSelector);

				expect(location).toHaveAttribute('aria-checked', 'true');

				location.focus();
				userEvent.keyboard('{space}');
				location = screen.getByTestId(locationSelector);

				expect(location).toHaveAttribute('aria-checked', 'true');
			});

			it('selects next/first location when hitting down/right arrow', () => {
				expect(location.getAttribute('aria-checked')).toBe(null);
				expect(nextLocation.getAttribute('aria-checked')).toBe(null);

				location.focus();
				userEvent.keyboard('{arrowright}');
				location = screen.getByTestId(locationSelector);
				nextLocation = screen.getByTestId(nextLocationSelector);

				expect(location).toHaveAttribute('aria-checked', 'false');
				expect(nextLocation).toHaveAttribute('aria-checked', 'true');
			});

			it('selects previous/last location when hitting up/left arrow', () => {
				expect(location.getAttribute('aria-checked')).toBe(null);

				location.focus();
				userEvent.keyboard('{arrowleft}');
				location = screen.getByTestId(locationSelector);
				previousLocation = screen.getByTestId(previousLocationSelector);

				expect(location).toHaveAttribute('aria-checked', 'false');
				expect(previousLocation).toHaveAttribute('aria-checked', 'true');
			});
		});
	});

	describe('Communication', () => {
		// Create element to attach component to it and avoid warnings when attached to document.body
		// https://stackoverflow.com/a/49025532/9826498
		const container = document.createElement('div');
		document.body.appendChild(container);

		const handleOnChange = jest.fn();

		beforeEach(() => {
			wrapper = render(
				<RadioSVGMap
					map={FakeMap}
					selectedLocationId="id1"
					onChange={handleOnChange}
				/>,
				{ attachTo: container }
			);
			location = screen.getByTestId(locationSelector);
			nextLocation = screen.getByTestId(nextLocationSelector);
		});

		afterEach(() => {
			wrapper.unmount();
			handleOnChange.mockClear();
		});

		it('selects initial location when id is provided', () => {
			expect(location).toHaveAttribute('aria-checked', 'true');
		});

		it('calls onChange handler when selecting location', () => {
			userEvent.click(nextLocation);

			expect(handleOnChange).toHaveBeenCalledWith(nextLocation);
		});

		it('does not call onChange handler when clicking on already selected location', () => {
			userEvent.click(location);

			expect(handleOnChange).toHaveBeenCalledTimes(0);
		});
	});

	describe('Rendering', () => {
		beforeAll(() => {
			// Mock ReactDOM to avoid error
			ReactDOM.findDOMNode = jest.fn(
				() => ({
					getElementsByTagName: jest.fn(() => ([]))
				})
			);
		});

		it('displays map with default props', () => {
			const component = renderer.create(<RadioSVGMap map={FakeMap} />);
			const tree = component.toJSON();

			expect(tree).toMatchSnapshot();
		});

		it('displays map with custom props', () => {
			const eventHandler = () => 'eventHandler';
			const component = renderer.create(
				<RadioSVGMap map={FakeMap}
					className="className"
					locationClassName="locationClassName"
					onLocationMouseOver={eventHandler}
					onLocationMouseOut={eventHandler}
					onLocationMouseMove={eventHandler}
					onLocationFocus={eventHandler}
					onLocationBlur={eventHandler}
					onChange={eventHandler}
					childrenBefore={<text>childrenBefore</text>}
					childrenAfter={<text>childrenAfter</text>}
				/>
			);
			const tree = component.toJSON();

			expect(tree).toMatchSnapshot();
		});
	});
});
