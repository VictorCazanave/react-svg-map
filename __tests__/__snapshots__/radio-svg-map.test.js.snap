// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioSVGMap component Rendering displays map with custom props 1`] = `
<svg
  aria-label="label"
  className="className"
  role="radiogroup"
  viewBox="viewBox"
  xmlns="http://www.w3.org/2000/svg"
>
  <text>
    childrenBefore
  </text>
  <path
    aria-checked={null}
    aria-label="name0"
    className="locationClassName"
    d="path0"
    id="id0"
    name="name0"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseMove={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    role="radio"
    tabIndex="0"
  />
  <path
    aria-checked={null}
    aria-label="name1"
    className="locationClassName"
    d="path1"
    id="id1"
    name="name1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseMove={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    role="radio"
    tabIndex="-1"
  />
  <path
    aria-checked={null}
    aria-label="name2"
    className="locationClassName"
    d="path2"
    id="id2"
    name="name2"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseMove={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    role="radio"
    tabIndex="-1"
  />
  <text>
    childrenAfter
  </text>
</svg>
`;

exports[`RadioSVGMap component Rendering displays map with default props 1`] = `
<svg
  aria-label="label"
  className="svg-map"
  role="radiogroup"
  viewBox="viewBox"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    aria-checked={null}
    aria-label="name0"
    className="svg-map__location"
    d="path0"
    id="id0"
    name="name0"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="radio"
    tabIndex="0"
  />
  <path
    aria-checked={null}
    aria-label="name1"
    className="svg-map__location"
    d="path1"
    id="id1"
    name="name1"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="radio"
    tabIndex="-1"
  />
  <path
    aria-checked={null}
    aria-label="name2"
    className="svg-map__location"
    d="path2"
    id="id2"
    name="name2"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="radio"
    tabIndex="-1"
  />
</svg>
`;
